docs/source/app-dev/parties-users.rst

```diff
@@ -0,0 +1,88 @@
+
+Parties and Users On a Daml Ledger
+##################################
+
+Identifying parties and users is an important part of building a workable Daml application. Recall these definitions from the :doc:`Getting Started Guide <getting-started/app-architecture.html>`:
+
+- **Parties** are unique across the entire Daml network. These must be allocated before you can use them to log in, and allocation results in a random-looking (but not actually random) string that identifies the party and is used in your Daml code. Parties are a builtin concept.
+
+- On each participant node you can create **users** with human-readable user ids. Each user can be associated with one or more parties allocated on that participant node, and refers to that party only on that node. Users are a purely local concept, meaning you can never address a user on another node by user id, and you never work with users in your Daml code; party ids are always used for these purposes. Users are also a builtin concept.
+
+This represents a change from earlier versions of Daml, and the differences between versions are discussed in more depth here.
+
+Parties in SDK 1.18 and Previous
+********************************
+
+First, let's recap how parties worked in SDK 1.18. Parties could be fully user-controlled via the party id hint, which could be specified on party allocation, e.g., via Daml Script’s allocatePartyWithHint. The call allocatePartyWithHint “Alice” (PartyIdHint “alice”) allocated a party with the display name Alice and the party id alice. If a party with the id alice already existed, the allocation failed. This had a few implications:
+
+- You could allocate the same party id even if you restarted your ledger.
+
+- Party ids were human-readable (provided you choose human-readable party id hints). Following the user with the display name Bob in create-daml-app was easy, because their party id was also Bob.
+
+- Sandbox (but not the Daml Driver for SQL 1.x) implicitly allocated parties. This meant that if you created a contract with Bob as an observer or signatory, and no party with the id Bob had been allocated before, a party with the id Bob would be created without further action on your part.
+
+Parties in SDK 2.0 and Subsequent
+*********************************
+
+In Daml 2.0 and later versions, when you allocate a party with a given hint Alice either in the sandbox or on a production ledger you will get back a party id like Alice::1220f2fe29866fd6a0009ecc8a64ccdc09f1958bd0f801166baaee469d1251b2eb72. The prefix before the double colon corresponds to the hint specified on party allocation. If the hint is not specified, it defaults to party-${randomUUID}. The suffix is the fingerprint of the public key that can authorize topology transactions for this party. Keys are generated randomly, so the suffix will look different locally and every time you restart Sandbox, you will get a different party id. This has a few new implications:
```

Comment:
What do you mean by "softly"? It should be possible to make links within the docs without breaking the build.

Reply:
> softly

so that the PDF link takes you to the appropriate section in the same PDF instead of the web.

The `doc` directive seems to break the build if the target isn't present; I'm not sure how well that holds up with the split build.

Reply:
Oh right. `:doc:` links should result in such "soft" links; for links to the canton parts, there's a trick we use to workaround the split by creating false targets in [docs/source/canton-refs.rst](https://raw.githubusercontent.com/digital-asset/daml/main/docs/canton-refs.rst).

The "real" target must exist on the Canton side, of course.